net/uring/netenc/encoding_test.go

```diff
@@ -0,0 +1,54 @@
+package netenc
+
+import (
+	"encoding/binary"
+	"testing"
+)
+
+func TestNtohs(t *testing.T) {
+	v := [2]byte{0xCD, 0xAB}
+	big := binary.BigEndian.Uint16(v[:])
+	if 0xABCD != Ntohs(big) {
+		t.Errorf("ntohs failed, want %v, got %v", 0xABCD, Ntohs(big))
+	}
+}
+
+func toBytes32(v uint32) [4]byte {
```

Comment:
This is fine, but out of curiosity, why not use encoding/binary here?

Reply:
oh I was thinking that this should work with the native endianness (I might be wrong), but binary encoding requires passing either little or big

Reply:
Silly me. The only way to do native endianness is with unsafe. So these tests are going to need to use unsafe. (That's OK; unsafe in tests is a different matter than unsafe not in tests.)

Something like (untested):

```go
func TestHton16(t *testing.T) {
  h := 0xABCD
  n := Hton16(h)
  got := *(*[4]byte)(unsafe.Pointer(&n))
  want := [4]byte{0xA, 0xB, 0xC, 0xD}
  if want != got {
    qt.Assert(want, qt.Equals, got)
  }
}
```